src/main/java/com/codacy/client/stash/client/auth/Parameter.java

```diff
@@ -0,0 +1,55 @@
+package com.codacy.client.stash.client.auth;
+
+/**
+ * Helper class for sorting query and form parameters that we need
+ */
+public class Parameter implements Comparable<Parameter> {
+    private final String key, value;
+
+    public Parameter(String key, String value) {
+        this.key = key;
+        this.value = value;
+    }
+
+    public String key() {
+        return key;
+    }
+
+    public String value() {
+        return value;
+    }
+
+    @Override
+    public int compareTo(Parameter other) {
+        int diff = key.compareTo(other.key);
+        if (diff == 0) {
+            diff = value.compareTo(other.value);
+        }
+        return diff;
+    }
+
+    @Override
+    public String toString() {
+        return key + "=" + value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method equals() has an NPath complexity of 24](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903589/issues/source?bid=2734487&fileBranchId=3159563#l37)